### ISSUE TITLE
Document job cleaning

### ISSUE THREAD

**Issue by jdcc** [vsivsi/meteor-job-collection#118]
First of all, thank you for this package, I really appreciate how communicative you have been with the users of JC - it's seriously impressive.

tl;dr - Can all `Job.jobStatusRemovable` jobs be safely removed even if they're to be repeated?
 
I'm working on a project that has jobs running very often and repeating forever, and as such, the job collection has grown quite large. I knew this would be the case, but I've been putting off writing a proper job cleaner because I fear I don't know enough about the underlying implementation.

I've read through #36 and #101, and understand and agree with your position on not supplying a default job cleaner, but without documentation it feels like an unnecessarily heavy lift for me (Step 1. Learn coffeescript...).

What I'm particularly confused about is which jobs need to continue existing in mongo so that jobs continue to repeat. When a job runs, does it schedule it's next repeat so that every completed job can be safely deleted, or, when polling, does the manager look for the most recently completed repeating job? The cleaner in the playground app doesn't directly address this (is it keeping the last five minutes of jobs for convenience or necessity?).

There have been a few questions about job cleaning now, so I think sticking something in the docs could be justified. As I see it, there are at least 3 approaches: writing a cleaning job, capping the job collection, or using lifecycle callbacks to have jobs remove themselves. Without digging in and understanding the full implementation, I worry that I can't implement any of them without potentially borking jobs.

Any pointers?

**Comment by vsivsi** [comment]
> When a job runs, does it schedule it's next repeat so that every completed job can be safely deleted, or, when polling, does the manager look for the most recently completed repeating job?

Yes.  The next repeat/iteration is a new job and it is written to Mongo before the finishing job is marked as "completed" in the database. however, if a job is "cancelled" or "failed" and you clean/remove it, all future potential repeats will be lost... 

> The cleaner in the playground app doesn't directly address this (is it keeping the last five minutes of jobs for convenience or necessity?).

It keeps them so that you can see a short history of finished jobs in the UI. It's a demo app, so if they were removed too quickly people would wonder if they finished at all!

Seems that you've done your homework on this. I'm always open to documentation PRs that improve the state of things. But I'm personally completely slammed until January, and so I need to really save the time I spend on JC for critical bugs and such.

Maybe you can "@" some other active users from the other threads in here and collaborate on some ideas / solutions. I'll be happy in a couple of months to implement any concrete ideas you guys come up with.  As you've already learned, this is an aspect of the system I intentionally left under-specified hoping to solicit feedback from users with actual experience using it in production.  I think that experience now exists, someone just needs to pull the ideas together.  If you can wait until 2016, I'm happy to do it, otherwise if I were you I'd hoist a flag and see if anyone else shows up to help.

**Comment by jdcc** [comment]
Thanks for the info - really helpful. With that information and the example from the playground app it's pretty trivial to roll one's own solution.

I'll put together a documentation PR at least outlining some options and pointing to resources.

**Comment by vsivsi** [comment]
Sounds good. Thanks!

**Comment by vsivsi** [comment]
Closing this after the above merge.
